vcd/resource_vcd_dnat.go

```diff
@@ -136,7 +150,7 @@ func resourceVcdDNATRead(d *schema.ResourceData, meta interface{}) error {
 	var found bool
 
 	for _, r := range e.EdgeGateway.Configuration.EdgeGatewayServiceConfiguration.NatService.NatRule {
-		if r.RuleType == "DNAT" &&
+		if r.RuleType == "DNAT" && r.GatewayNatRule.Interface.Name == d.Get("network_name") &&
```

Comment:
`network_name` may be ommited because it is optional.

Reply:
Addressed